sql/core/src/test/scala/org/apache/spark/sql/execution/columnar/InMemoryColumnarQuerySuite.scala

```diff
@@ -390,4 +393,23 @@ class InMemoryColumnarQuerySuite extends QueryTest with SharedSQLContext {
     }
   }
 
+  test("InMemoryTableScanExec should return currect output ordering and partitioning") {
+    val ds1 = Seq((0, 0), (1, 1)).toDS
+      .repartition(col("_1")).sortWithinPartitions(col("_1")).persist
+    val ds2 = Seq((0, 0), (1, 1)).toDS
+      .repartition(col("_1")).sortWithinPartitions(col("_1")).persist
+    val joined = ds1.joinWith(ds2, ds1("_1") === ds2("_1"))
```

Comment:
Because the two datasets cached have the same logical plan, it is a self-join actually.